modules/ROOT/pages/6.4-release-notes.adoc

```diff
@@ -0,0 +1,1252 @@
+
+In {productname} 6.4, pressing backspace in an empty line now preserves formatting in a previous empty line.
+
+// https://notion.so/tinycloud/Four-words-to-live-by-96e307110aaa4137b94f9ba12874d480
+
+=== Pressing enter inside the inputfontsize input would not move the focus back into the editor content
+
+// https://ephocks.atlassian.net/browse/TINY-9598
+During development for the new `fontsizeinput` toolbar item, an issue was found that by pressing enter inside the input field would not move the `focus` back into the {productname} editor body. 
+
+As a consequence, any changes made to the font size within the new toolbar input field, would result in the user having no way to use the new font size within the editor.
+
+After some improvements to the `fontsizeinput` toolbar item, pressing enter in the input field now sets the focus back to the {productname} editor viewport, which allows the new font size to be used. 
+
+In {productname} 6.4, pressing enter inside the `inputfontsize` input now returns the focus back to the editor content body.
+// https://notion.so/tinycloud/Four-words-to-live-by-96e307110aaa4137b94f9ba12874d480
+
+
+[[additions]]
+== Additions
+
+{productname} 6.4 also includes the following additions:
+
+=== New tree component that can be used in dialog body panel
+
+// https://ephocks.atlassian.net/browse/TINY-9532
+A new tree component has been created to support the *Advanced Template* premium plugin that works in conjuction with a new tree dialog body panel.
+
+This tree component is limited to *one* level and is used in the dialog body panel to allow users to organize templates into levels for better UI. The tree component is designed to create a visual representation of the hierarchical structure for the template system, which allows users to easily identify the various templates and levels within each catagory, giving users the ability to quickly and easily navigate through the templates, and select the right one for their needs.
+
+This new tree component is an important addition to the new Premium Plugin: *Advanced Templates*, which is now available as a premium plugin with editors running {productname} v6.4+.
+
+// https://notion.so/tinycloud/Four-words-to-live-by-96e307110aaa4137b94f9ba12874d480
+
+=== renderUI property in the Theme type can now return a Promise <RenderResult> instead of RenderResult
+
+// https://ephocks.atlassian.net/browse/TINY-9556
+In earlier versions of {productname}, the properties set for the `renderUI`, was causing issues with the theme rendering load times. As a result {productname} 6.4 added fixes to the `renderUI` properties to correct this issue.
+
+The changes saw the introduction of a new `renderUI` property called `Promise<RenderResult>` , which allows the current theme type to return a `Promise<RenderResult>` instead of the origional  `RenderResult`.
+
+In {productname} 6.4, the `renderUI` property in the Theme type, now returns a `Promise <RenderResult>`.
```

Comment:
This makes it sound far more complicated than it is. And "properties" is not at all the correct way to describe the change.

```suggestion
In earlier versions of {productname} the `renderUI` function which initiates theme rendering did not wait for the CSS to load. This could cause race conditions with other parts of the editor or customer integrations. As a result {productname} 6.4 now waits for a CSS load event before returning from the `renderUI` function.

This process necessitates changing the return type of the `renderUI` function; it now returns a `Promise<RenderResult>`. The type is `Promise<RenderResult> | RenderResult` for compatibility, but the silver theme always returns a promise.

This change should have no impact on customer integrations, even those implementing custom themes.
```

Reply:
This suggestion has been added and merged into the documentation. TY